tests/device_fixtures.py

```diff
@@ -407,22 +408,28 @@ async def _discover_update_and_close(ip, username, password) -> Device:
     return await _update_and_close(d)
 
 
-async def get_device_for_fixture(fixture_data: FixtureInfo) -> Device:
+async def get_device_for_fixture(
+    fixture_data: FixtureInfo, *, verbatim=False, update=True
+) -> Device:
     # if the wanted file is not an absolute path, prepend the fixtures directory
 
     d = device_for_fixture_name(fixture_data.name, fixture_data.protocol)(
         host="127.0.0.123"
     )
     if fixture_data.protocol in {"SMART", "SMART.CHILD"}:
-        d.protocol = FakeSmartProtocol(fixture_data.data, fixture_data.name)
+        d.protocol = FakeSmartProtocol(
+            fixture_data.data, fixture_data.name, verbatim=verbatim
+        )
     elif fixture_data.protocol == "SMARTCAMERA":
-        d.protocol = FakeSmartCameraProtocol(fixture_data.data, fixture_data.name)
+        d.protocol = FakeSmartCameraProtocol(
+            fixture_data.data, fixture_data.name, verbatim=verbatim
+        )
     else:
-        d.protocol = FakeIotProtocol(fixture_data.data)
+        d.protocol = FakeIotProtocol(fixture_data.data, verbatim=verbatim)
 
     discovery_data = None
     if "discovery_result" in fixture_data.data:
-        discovery_data = {"result": fixture_data.data["discovery_result"]}
+        discovery_data = fixture_data.data["discovery_result"]
```

Comment:
Found a test framwork bug here, `update_from_discover_info(discovery_data)` should not include the `result` parent tag.